### PR TITLE
Python transformations doc update

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Updated transformations doc to add support for Python

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/features/transformations/index.mdx [hunk 1]
```diff
@@ -1,149 +1,169 @@
...
 ## Adding a transformation
 
 To add a new transformation in the RudderStack dashboard, follow these steps:
 
-1. Log into the [RudderStack dashboard](https://app.rudderstack.com/).
-2. Go to **Enhance** > **Transformations**, as shown:
+1. Log in to the [RudderStack dashboard](https://app.rudderstack.com/).
+2. Go to **Enhance** > **Transformations** and click **New Transformation**.
+3. Name your transformation and add an optional description.
+4. In the **Transformation** window, select the language to write your transformations, as shown:
```
Comment (ishachopra0401) [inline docs/features/transformations/index.mdx:81]:
```suggestion
4. In the **Transformation** window, select the language to write your transformation, as shown:
```

File: docs/features/transformations/index.mdx [hunk 2]
```diff
@@ -1,149 +1,169 @@
...
 
 <img src="../../assets/features/adding-a-transformation-1.png" alt="Adding a Transformation" />
 
-3. Click **New Transformation**.
-4. Add a name for your transformation and an optional description.
-5. Next, add your JavaScript function in the **Transformation** window, as shown:
+5. Add your transformation function in the , as shown:
```
Comment (ishachopra0401) [inline docs/features/transformations/index.mdx:85]:
```suggestion
5. Add your transformation function, as shown:
```

File: docs/features/transformations/index.mdx [hunk 3]
```diff
@@ -1,149 +1,169 @@
...
 
 <img src="../../assets/features/adding-a-transformation-2.png" alt="Adding a Transformation" />
 
-<div class="successBlock">
+<div class="infoBlock">
 You can also add other functions and call them from within <code class="inline-code">transformEvent</code>.
```
Comment (ishachopra0401) [inline docs/features/transformations/index.mdx:90]:
```suggestion
You can also add other functions and call them from within the <code class="inline-code"><Link to="#transformevent-function">transformEvent</Link></code> function.
```

File: docs/features/transformations/cloud-mode-transformations.mdx [hunk 4]
```diff
@@ -3,28 +3,65 @@ title: "Transformations in Cloud Mode"
 description: Use transformations with cloud mode destinations.
 ---
 
-When you <Link to="#adding-a-transformation">add a transformation</Link> and connect it to a destination supporting <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>, RudderStack:
+When you <Link to="/features/transformations/#adding-a-transformation">add a transformation</Link> and connect it to a <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link> destination, RudderStack does the following:
```
Comment (ishachopra0401) [inline docs/features/transformations/cloud-mode-transformations.mdx:6]:
```suggestion
When you <Link to="/features/transformations/#adding-a-transformation">add a transformation</Link> and connect it to a <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link> supported destination, RudderStack does the following:
```

File: docs/features/transformations/device-mode-transformations.mdx [hunk 5]
```diff
@@ -3,12 +3,15 @@ title: "Transformations in Device Mode"
 description: Use transformations with device mode destinations.
 ---
 
-RudderStack's Device Mode Transformations feature enables you to use the powerful user transformations with the [device mode-only destinations](https://github.com/rudderlabs/rudder-sdk-js/tree/production/integrations), for example, Firebase, Hotjar, etc.
+You can use RudderStack's Transformations feature with the [device mode destinations](https://github.com/rudderlabs/rudder-sdk-js/tree/production/integrations) like Firebase, Hotjar, etc.
```
Comment (ishachopra0401) [inline docs/features/transformations/device-mode-transformations.mdx:6]:
```suggestion
You can use RudderStack's Transformations feature with the [device mode supported destinations](https://github.com/rudderlabs/rudder-sdk-js/tree/production/src/integrations) like Firebase, Hotjar, etc. 
```

File: docs/features/transformations/device-mode-transformations.mdx [hunk 6]
```diff
@@ -3,12 +3,15 @@ title: "Transformations in Device Mode"
...
+To learn more about the use cases of device mode transformations, refer to the <Link to="/features/transformations/#use-case">Use case</Link> section.
 
-<div class="warningBlock">
-Only the <Link to="/sources/event-streams/sdks/rudderstack-android-sdk/">Android</Link>, <Link to="/sources/event-streams/sdks/rudderstack-ios-sdk/">iOS</Link>, and <Link to="/sources/event-streams/sdks/rudderstack-javascript-sdk/">JavaScript</Link>  SDKs support the Transformations feature to send events to the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link> destinations.
+<div class="infoBlock">
+<ul>
+<li>Only the <Link to="/sources/event-streams/sdks/rudderstack-javascript-sdk/">JavaScript</Link>, <Link to="/sources/event-streams/sdks/rudderstack-android-sdk/">Android</Link>, and <Link to="/sources/event-streams/sdks/rudderstack-ios-sdk/">iOS</Link> SDKs support device mode transformations.</li>
+<li><strong>You can write device mode transformations only in JavaScript</strong>. Python support will be added soon.</li>
```
Comment (ishachopra0401) [inline docs/features/transformations/device-mode-transformations.mdx:13]:
```suggestion
<li><strong>Currently, RudderStack supports writing device mode transformations only in JavaScript.</strong></li>
```

File: docs/features/transformations/faq.mdx [hunk 7]
```diff
@@ -5,9 +5,17 @@ description: Answers to some of the commonly asked questions related to RudderSt
 
 ## Transformations
 
+#### Is there a limit on the number of transformations I can create and use?
+
+The RudderStack Cloud **Free** and **Starter** plans let you create 5 JavaScript transformations. However, there is no limit on this number in the **Growth** and **Enterprise** plans. Refer to the <a href="https://www.rudderstack.com/pricing/">Pricing</a> page for more information.
```
Comment (ishachopra0401) [inline docs/features/transformations/faq.mdx:10]:
```suggestion
The RudderStack Cloud **Free** and **Starter** plans let you create 5 JavaScript transformations. However, you can create unlimited transformations in the **Growth** and **Enterprise** plans. Refer to the <a href="https://www.rudderstack.com/pricing/">Pricing</a> page for more information.
```

File: docs/features/transformations/faq.mdx [hunk 8]
```diff
@@ -5,9 +5,17 @@ description: Answers to some of the commonly asked questions related to RudderSt
...
+
+<div class="infoBlock">
+Python support for creating transformations is only available for the <strong>Growth</strong> and <strong>Enterprise</strong> plans.
```
Comment (ishachopra0401) [inline docs/features/transformations/faq.mdx:13]:
```suggestion
RudderStack supports creating transformations in Python only in the <strong>Growth</strong> and <strong>Enterprise</strong> plans.
```

File: docs/features/transformations/using-libraries.mdx [hunk 9]
```diff
@@ -106,10 +106,10 @@ getPrice(ev)
 
 ## Deleting a library
 
-To delete a library, go to **Enhance** > **Transformations** and click the **Libraries** tab. Click the **Delete** button next to the library you want to delete, as shown:
```
Comment (Jayachand) [inline docs/features/transformations/using-libraries.mdx:109]: Library can also be deleted. why the removal of this statement?

File: docs/features/transformations/index.mdx [hunk 10]
```diff
@@ -1,149 +1,169 @@
...
-RudderStack supports the Transformations feature across your <Link to="/sources/event-streams/">Event Streams</Link>, <Link to="/sources/extract/">Cloud Extract</Link>, and <Link to="/sources/reverse-etl/">Reverse ETL</Link> pipelines in both the cloud and device modes. Refer to the <Link to="/features/transformations/device-mode-transformations">Device Mode Transformations</Link> and <Link to="/features/transformations/cloud-mode-transformations">Cloud Mode Transformations</Link> documentation for more details on using transformations in each <Link to="/destinations/rudderstack-connection-modes/">connection mode</Link>.
-
-The below table summarizes the availability of Transformations feature across different [pricing plans](https://www.rudderstack.com/pricing/):
-
-| Connection Mode | Free       | Pro   | Enterprise    |
-| :------------------ | :------------ | :------------ | :------------ |
-| **Cloud mode**      | Upto 3 | ✅  | ✅   |
-| **Device mode**     | ❌    | ❌      | ✅        |
+RudderStack's **Transformations** feature lets you write custom functions to implement specific use-cases on your event data, like event filtering, sampling, aggregation, enrichment, and more.
```
[the remark this answers is not in the document]
Comment (Jayachand) [reply docs/features/transformations/index.mdx:6]: +1 to bullet point structure

File: docs/features/transformations/cloud-mode-transformations.mdx [hunk 11]
```diff
@@ -36,6 +73,25 @@ export function transformEvent(event, metadata) {
...
+    <TabPanel>
+<span>
+
+```python
+def transformEvent(event, metadata):
+    meta = metadata(event)
+    event['sourceId'] = meta['sourceId']
+    
+    log("Event Name is", event['event'], ";", "Message ID is", event['messageId'])
+    log("Source ID is", meta['sourceId']);
```
Comment (Jayachand) [inline docs/features/transformations/cloud-mode-transformations.mdx:87]: remove `;` at the end

File: docs/features/transformations/faq.mdx [hunk 12]
```diff
@@ -5,9 +5,17 @@ description: Answers to some of the commonly asked questions related to RudderSt
...
+The RudderStack Cloud **Free** and **Starter** plans let you create 5 JavaScript transformations. However, you can create unlimited transformations in the **Growth** and **Enterprise** plans. Refer to the <a href="https://www.rudderstack.com/pricing/">Pricing</a> page for more information.
+
+<div class="infoBlock">
+RudderStack supports Python transformations only in the <strong>Growth</strong> and <strong>Enterprise</strong> plans.
+</div>
+
 #### Are there any sample user transformations I can refer to?
 
-Yes, you can refer to the [Sample RudderStack Transformations](https://github.com/rudderlabs/sample-user-transformers) GitHub repository.
+Yes - refer to the [Sample RudderStack Transformations](https://github.com/rudderlabs/sample-user-transformers) GitHub repository.
```
Comment (Jayachand) [inline docs/features/transformations/faq.mdx:18]: just `Refer to` would be better

File: docs/features/transformations/index.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/features/transformations/index.mdx]: Reverted!

**Review by Jayachand — APPROVED** [review]

**Pull request merged**